File: Project4/cpp/src/main.cpp
#include <iostream>
#include "metamodel.h"
#include "solver.h"

int main(int argc, char const *argv[]) {
    // Read the parameter path from commandline
    // Default to "parameters.json"
    std::string parameterPath = "parameters.json";
    if (argc == 2)
        parameterPath = argv[1];
    parameterPath = "../data/" + parameterPath;

    try{
        Metamodel model = Metamodel();
        model.read(parameterPath);
        solveSystem(model);
    } catch (std::exception& ex) {
        std::cerr << "Error: " << ex.what() << std::endl;
    }

    return 0;
}

File: Project4/julia/src/main.jl
include("ising.jl")
 # include("metamodel.jl")
using .Ising
# using .Metamodel

function run()
    model = Metadata(50, 1000, 1.0, :random, "../../data/"; seed=1334)
    # model.J = 1
    ising!(model, :serial)
end

run()

